### PR TITLE
core: use -default-* flags

### PR DESCRIPTION
**Pull request by Javlopez**
enable flags with --default prefix for
 
- ` --log-to-stderr`
- `--mon-cluster-log-to-stderr`
- `--err-to-stderr`
- `--log-stderr-prefix`

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves https://github.com/rook/rook/issues/11429

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [ ] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: pkg/operator/ceph/test/containers.go [hunk 1]
```diff
@@ -58,10 +58,10 @@ func (ct *ContainersTester) AssertArgsContainCephRequirements() {
 			continue // don't consider containers that aren't Ceph commands
 		}
 		requiredFlags := []string{
-			"--log-to-stderr=true",
-			"--err-to-stderr=true",
-			"--mon-cluster-log-to-stderr=true",
-			"--log-stderr-prefix=debug ",
+			"--default-log-to-stderr=true",
```
Comment (parth-gr) [inline pkg/operator/ceph/test/containers.go:61]: I am not sure how users can override these settings if they so desire.
As requested

Comment (Javlopez) [reply pkg/operator/ceph/test/containers.go:61]: Hi, as I understand in the issue description, some users wants to use options with `--default-*`

> Some users have wished to change these options for whatever reasons in the past. With `--default-*` versions, Rook can set CLI options that will ensure the average user will still get all logs to stderr as they would expect, but advanced users could enjoy different behavior.

This change should be enough, but  I'm not sure if I'm missing something

**Review by travisn — APPROVED** [review]
LGTM, @BlaineEXE please review before we merge

**Review by travisn — APPROVED** [review]
LGTM, @BlaineEXE please review before we merge

**Comment by travisn** [comment]
@Javlopez Was this accidentally closed?

**Comment by Javlopez** [comment]
Hi @travisn yes, I closed due I recreate my fork, sorry for that here is the new PR which contains the same changes https://github.com/rook/rook/pull/12302

**Comment by travisn** [comment]
> Hi @travisn yes, I closed due I recreate my fork, sorry for that here is the new PR which contains the same changes #12302

I see. In the future, you may just need to create a new local clone, and no need to create your fork again.

**Pull request closed without merging**